src/aircraft_cmd.cpp

```diff
@@ -1092,6 +1098,15 @@ static bool AircraftController(Aircraft *v)
 			z = GetAircraftFlightLevel(v);
 		}
 
+		/* Some airports (like a rotated intercontinental) are non-rectangular and their primary (north-most) tile might not be at the same height as the runway.
+		 * Therefore, as the hangar must be on flat ground (and we must have a hangar, as we're landing/braking and so not a helicopter), use that as the z position instead. */
+		int airport_z = v->z_pos;
```

Comment:
would you mind making the length of these comments shorter, by hitting <enter> key once in a while? It became a bit lengthy, that even GitHub has issues rendering it nicely on my 2k screens :P

Reply:
Additionally, the comment was not obvious to me what you meant. What I was missing, is that in this case, the north-most tile is not part of the airport. The rest of the text sounds like a mad man mumbling, but I think I understand it like this:

```
/* NewGRF airports (like a rotated intercontinental from OpenGFX+Airports) can be non-rectangular
 * and their primary (north-most) tile does not have to be part of the airport.
 * As such, the height of the primary tile can be different from the rest of the airport.
 * Given we are landing/breaking, and as such are not a helicopter, we know that there has to be a hangar.
 * We also know that the airport itself has to be completely flat (otherwise it is not a valid airport).
 * Therefore, use the height of this hangar to calculate our z-value. */
```